### ISSUE TITLE
Use deploy-rs for home manager deployments, too

### ISSUE THREAD

**Issue by blaggacao** [divnix/digga#81]
deploy-rs is versatile enough and [has solid primitives](https://github.com/serokell/deploy-rs/blob/364ef202e400e4c941e18833ca38fa848ac5a148/flake.nix#L98)

after https://github.com/serokell/deploy-rs/issues/99 it also will be the superior option for local deploys.

/cc @Pacman99

**Comment by Pacman99** [comment]
I don't think this should be included in `mkDeployNodes` because we don't know what hosts a user would want their portable home configuration to be sent to and just sending it to all hosts could become quite annoying.
And including special casing in `mkDeployNodes` is pointless because we can't cover all the possibles except for the most general.

While trying to implement this feature I also realized that `mkDeployNodes` can itself be quite counter-productive. I think it might just be a good idea to remove it as it is this half integration without giving any customization options and it makes so many assumptions.

**Comment by blaggacao** [comment]
There has been a stalled upstream-discussion that pointed out that hm itself is ill-designed here: a host-less user is to be system scoped.
